referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
+
+  var subresourceInvoker = {
+    "a-tag": queryLink,
+    "area-tag": queryAreaLink,
+    "fetch-request": queryFetch,
+    "iframe-tag": queryIframe,
+    "img-tag":  queryImage,
+    "script-tag": queryScript,
+    "worker-request": queryWorker,
+    "xhr-request": queryXhr
+  };
+
+  var pathForSubresource = SPEC_JSON["subresource_path"];
+
+  var referrerUrlResolver = {
+    "omitted": function() {
+      return undefined;
+    },
+    "origin": function() {
+      return location.protocol + "//" + location.hostname +
+             normalizePort(location.port) + "/";
```

Comment:
Origins usually don't have trailing slashes. Are we sending a trailing slash in the header?

Hrm. It looks like we do. Why do we do that? I know I wrote the code and the spec, but I disagree with myself. :) We don't add a `/` for the `Origin:` header, so we shouldn't add it here either.

Let's discuss in https://github.com/w3c/webappsec/issues/382 and come back to this later.